### PR TITLE
Support for IonTorrent data

### PR DESCRIPTION
**Pull request by jtangrot**
Added support for (single-ended) IonTorrent reads. Parameters changed compared to Illumina are dada(..., HOMOPOLYMER_GAP_PENALTY=-1, BAND_SIZE=32) and filterAndTrim(..., trimLeft=15) - all according to recommendations at https://benjjneb.github.io/dada2/faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
I'm not quite sure where to document this.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/ampliseq.nf [hunk 1]
```diff
@@ -40,6 +40,7 @@ if (params.qiime_ref_taxonomy && !params.skip_taxonomy && !params.classifier) {
  */
 
 single_end = params.pacbio ? true : params.single_end
+single_end = params.iontorrent ? true : params.single_end
```
Comment (d4straub) [inline workflows/ampliseq.nf:43]: This doesn't work as intended. This line will overwrite `single_end = true` with `single_end = false` if `pacbio = true` and `iontorrent = false`. Thats why the PacBio test fails at the moment.

Comment (jtangrot) [reply workflows/ampliseq.nf]: Of course, thanks! Should be fixed now.

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ Re-wrote whole pipeline in nextflow [DSL2](https://www.nextflow.io/docs/latest/d
 * [#275](https://github.com/nf-core/ampliseq/pull/275) - Read count summary
 * [#274](https://github.com/nf-core/ampliseq/pull/274) - `--skip_qiime` to prevent any steps that are executed with QIIME2
 * [#272](https://github.com/nf-core/ampliseq/pull/272) - `--cut_its` to cut ASV sequence to ITS region before performing taxonomic classification with DADA2
+* Added support for IonTorrent data
```
Comment (d4straub) [inline CHANGELOG.md:23]:
```suggestion
* [#272](https://github.com/nf-core/ampliseq/pull/280) - Added support for IonTorrent data
```

File: workflows/ampliseq.nf [hunk 3]
```diff
@@ -117,9 +129,10 @@ multiqc_options.args       += params.multiqc_title ? " --title \"$params.multiqc
 
 def dada2_taxonomy_options  = modules['dada2_taxonomy']
 dada2_taxonomy_options.args += params.pacbio ? ", tryRC = TRUE" : ""
+dada2_taxonomy_options.args += params.iontorrent ? ", tryRC = TRUE" : ""
 
 def dada2_addspecies_options  = modules['dada2_addspecies']
-dada2_addspecies_options.args += params.pacbio ? ", tryRC = TRUE" : ""
+dada2_addspecies_options.args += params.iontorrent ? ", tryRC = TRUE" : ""
```
Comment (d4straub) [inline workflows/ampliseq.nf:135]: Why are you removing `tryRC = TRUE` for `pacbio`? `tryRC = TRUE` is used for `pacbio` and `iontorrent` in `dada2_taxonomy_options`, why only for `iontorrent` but not `pacbio` in `dada2_addspecies_options`?

**Review by d4straub — CHANGES_REQUESTED** [review]
Looks pretty good!

Could you please add `test_iontorrent` to the [ci.yml line 53](https://github.com/jtangrot/ampliseq/blob/c133406540d30dc6fa9bf752d3a2a3b0c8b6c475/.github/workflows/ci.yml#L53) so that the test is automatically started on PRs.

File: workflows/ampliseq.nf (hunk 3 above)
Comment (jtangrot) [reply workflows/ampliseq.nf]: That was not intentional, thanks for spotting it!

**Comment by d4straub** [comment]
> Added support for (single-ended) IonTorrent reads. Parameters changed [...]
> I'm not quite sure where to document this.

All parameters/options for `dada2_filtntrim` and `dada2_denoising` are captured in files ([1](https://github.com/nf-core/ampliseq/blob/36961007d68406749c9be4d63f89e9d26685e73b/modules/local/dada2_filtntrim.nf#L49),[2](https://github.com/nf-core/ampliseq/blob/36961007d68406749c9be4d63f89e9d26685e73b/modules/local/dada2_denoising.nf#L60)) and published in `results/dada2/args/` also described in [the docs](https://github.com/nf-core/ampliseq/blob/dev/docs/output.md#dada2) as 

> dada2/args/: Directory containing all parameters for DADA2 steps.

**Review by d4straub — APPROVED** [review]
Great!

**Pull request merged**